Sources/AppcuesKit/Presentation/ExperienceRendering/ExperienceStateMachine+State.swift

```diff
@@ -135,11 +162,11 @@ extension ExperienceStateMachine.Transition {
     static func fromIdlingToBeginningExperience(_ experience: ExperienceData) -> Self {
         // handle errors that occurred prior to starting the experience - such as flow deserialization issues
         if let error = experience.error, !error.isEmpty {
-            return .init(toState: nil, sideEffect: .error(.experience(experience, error), reset: true))
+            return .init(toState: .idling, sideEffect: .error(.experience(experience, error)), resetObservers: true)
```

Comment:
we no longer need to worry about using `nil` here and the `reset: true`, since we explicitly control observer maintenance on the transition via the new `resetObservers` flag. This allows removal of the logic to always reset on the move to `.idling` from before - as that is problematic when the new `.failing` state needs to end and process a new `.startExperience`. In general, I think it is more straightforward to handle the reset explicitly in each transition as desired, so I've updated all such spots.